ui/pages/confirmations/hooks/useCurrentConfirmation.ts

```diff
@@ -47,29 +49,37 @@ const useCurrentConfirmation = () => {
         ({ id: confirmId }) => confirmId === paramsTransactionId,
       );
     }
+
     if (!pendingConfirmation) {
       if (!latestPendingConfirmation) {
         setCurrentConfirmation(undefined);
         return;
       }
       pendingConfirmation = latestPendingConfirmation;
     }
+
     if (pendingConfirmation.id !== currentConfirmation?.id) {
       const unconfirmedTransaction =
         unconfirmedTransactions[pendingConfirmation.id];
       if (!unconfirmedTransactions) {
         setCurrentConfirmation(undefined);
         return;
       }
-      if (
-        pendingConfirmation.type !== ApprovalType.PersonalSign &&
-        pendingConfirmation.type !== ApprovalType.EthSignTypedData &&
-        unconfirmedTransaction.type !== TransactionType.contractInteraction
-      ) {
+
+      const isConfirmationRedesignType =
+        REDESIGN_APPROVAL_TYPES.find(
+          (type) => type === pendingConfirmation?.type,
+        ) ||
+        REDESIGN_TRANSACTION_TYPES.find(
+          (type) => type === unconfirmedTransaction?.type,
+        );
```

Comment:
May be extract above to a util method, it used here also: https://github.com/MetaMask/metamask-extension/pull/24182/files#diff-e78ba84d4b27f8a33233149f7c076f773341ab940a409651508f75a2af6e67eeR263

Reply:
Thanks @jpuri. I did consider this. I'm okay repeating code 2-3 before DRYing. We could create a util method if we feel strong about this 

Q: how would you structure the util method with the two different types to check? particularly, would the idea be to allow 2 different params, one for each type?